Create alphaday.md

Mikael337 opened the pull request:
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (DOT, USDC & USDT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

Mikael337 commented:
Hello @keeganquigley, I have reopened this pull request since we have now fully open sourced Alphaday here: https://github.com/AlphadayHQ/alphaday

I request that you please reconsider the application.
Previous Application for reference: https://github.com/w3f/Grants-Program/pull/2285

keeganquigley reviewed (changes requested):
Thanks @Mikael337 happy to reconsider. I have a few initial comments:

- The grants program focuses mainly on the technical implementation, so while we can fund the development itself, we can't fund things like _Youtube stream containing educational Polkadot/Kusama/Web3 content_ or the podcast. Could you please remove these from the deliverables?
- The mandatory deliverables 0a. - 0d. are missing. Can you please add these as shown in the [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality)?

Mikael337 commented:
Hello @keeganquigley, I've updated the file with the newly requested changes.
Please let me know if other changes are required.

keeganquigley commented on applications/alphaday.md:
```diff
@@ -0,0 +1,195 @@
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | Not applicable to Alphaday, as we have no Substrates nodes or sending functionality on the app |
```
Documentation is always required for development milestones, regardless of the tech stack. At the very least a GH readme, but it's nice to see user-facing documentation as well, for front-end deliverables.

keeganquigley commented on applications/alphaday.md:
```diff
@@ -0,0 +1,195 @@
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | Not applicable to Alphaday, as we have no Substrates nodes or sending functionality on the app |
+| 0c. | Testing Guide | We can provide a document as a guide to give a bit more detail on how to test a completed dashboard's functionality. It will be a reference to the points listed below (1-14) |
```
We require unit tests as well, at least for the Python-Django backend. (But it's also nice to see Cypress tests or something similar for React front-ends). Would you be willing to include these?

keeganquigley commented on applications/alphaday.md:
```diff
@@ -0,0 +1,195 @@
+| 0a. | License | MIT |
+| 0b. | Documentation | Not applicable to Alphaday, as we have no Substrates nodes or sending functionality on the app |
+| 0c. | Testing Guide | We can provide a document as a guide to give a bit more detail on how to test a completed dashboard's functionality. It will be a reference to the points listed below (1-14) |
+| 0d. | Docker | The dashboard would be live and visible to anyone that accesses the url. As such, a Docker file would be deemed unnecessary. |
```
It's fine to go ahead and remove the Docker deliverable, as long as we can easily spin it up locally with Yarn. Maybe you could denote that here.

keeganquigley commented on applications/alphaday.md:
```diff
@@ -0,0 +1,195 @@
+
+### Milestone 2 — Maintenance & Curation of Dashboard
+
+- **Estimated duration:** Indefinitely
```
We won't be able to approve an unlimited estimated duration, could we maybe allocate 2 months here for now, since that's the timeline written in future plans.

keeganquigley reviewed (changes requested):
In addition to my above comments, we also require mockups for the front-end if you could include those too. Thanks!

Mikael337 commented:
Hello @keeganquigley ,

Thank you for the feedback. 

I spoke to the devs in our team and we are not able to provide:
- Mockups for the front-end
- Unit tests
- Documentation for front-end deliverables

The reasons varied.

I have gone ahead and removed the Docker file from the milestones and changed the duration of Milestone 2 to 2 months, as per your request.

Please let me know if we can still proceed without providing the 3 listed points above, or if there is a workaround somehow.

keeganquigley commented:
Thanks for your response @Mikael337, and for talking with your team.

I know this grant is rather small, so it makes sense to be small in scope. Therefore I understand it might not have certain components. Unfortunately, the above-mentioned deliverables are mandatory, so it would be unfair to other projects if we were to accept this one without them. 

The lack of mockups is the one item that could probably be forgiven at the current price. However, the lack of unit tests perplexes me because all code should generally be tested, with perhaps one exception for code written at a hackathon. Documentation is also something we have to have, otherwise how would we know how to run and test it?

Therefore, I personally won't approve the application; however, I will mark it as ready for review in case other committee members disagree.

semuelle reviewed (changes requested):
Why are you unable to provide unit tests? Any decent open source project should have thorough testing where applicable. If not, I don't think we should support it.

keeganquigley commented:
@Mikael337 I will go ahead and close this application due to the reasons noted above. If you are planning on addressing these concerns, please let me know and I will re-open it.
